### PR TITLE
Refactor ArgInNoCallerFunction, isDeadFunction and isNoCallerFunction

### PR DESCRIPTION
**Pull request by JasonZhongZexin**

### CODE REVIEW

File: include/Util/SVFModule.h [hunk 1]
```diff
@@ -209,6 +213,20 @@ class SVFModule
...
+    {
+        return argInNoCallerFunction;
+    }
+
+    inline Set<const Function*> getIsDeadFunction()
+    {
+        return isDeadFunction;
+    }
+
+    inline Set<const Instruction*> getIsReturn(){
```
Comment (mbarbar) [inline include/Util/SVFModule.h:226]: Return const reference for these -- performance.

Comment (mbarbar) [reply include/Util/SVFModule.h:226]: Not just performance, correctness.

File: include/Util/SVFModule.h [hunk 2]
```diff
@@ -209,6 +213,20 @@ class SVFModule
...
+        return isReturns;
+    }
+
```
Comment (mbarbar) [inline include/Util/SVFModule.h:229]: Add, `addReturn`, `addDeadFunction`, etc. functions.

File: lib/SVF-FE/SymbolTableBuilder.cpp [hunk 3]
```diff
@@ -206,6 +209,24 @@ void SymbolTableBuilder::collectNullPtrBlackholeSyms(const Value *val)
...
+void SymbolTableBuilder::collectDeadFunction(const Function * fun)
+{
+    if (LLVMUtil::isDeadFunction(fun))
+        symInfo->getModule()->getIsDeadFunction().insert(fun);
+}
+
+void SymbolTableBuilder::collectReturnInst(const Instruction *inst)
+{
+    if (LLVMUtil::isReturn(inst))
+        symInfo->getModule()->getIsReturn().insert(inst);
```
Comment (mbarbar) [inline lib/SVF-FE/SymbolTableBuilder.cpp:227]: `symInfo->getModule()->addReturnInst()`.

Inside, `addReturnInst()` -- update the set.

File: lib/SVF-FE/SymbolTableBuilder.cpp [hunk 4]
```diff
@@ -95,13 +96,15 @@ void SymbolTableBuilder::buildMemModel(SVFModule* svfModule)
...
             collectSym(&*I);
+            collectArgInNoCallerFunction(&*I);
         }
 
         // collect and create symbols inside the function body
         for (inst_iterator II = inst_begin(*fun), E = inst_end(*fun); II != E; ++II)
         {
             const Instruction *inst = &*II;
             collectSym(inst);
+            collectReturnInst(inst);
```
Comment (yuleisui) [inline lib/SVF-FE/SymbolTableBuilder.cpp:107]: move to `else if (const ReturnInst *ret = SVFUtil::dyn_cast<ReturnInst>(inst))`

File: lib/SVF-FE/SymbolTableBuilder.cpp [hunk 5]
```diff
@@ -206,6 +209,24 @@ void SymbolTableBuilder::collectNullPtrBlackholeSyms(const Value *val)
         symInfo->blackholeSyms.insert(val);
 }
 
+void SymbolTableBuilder::collectArgInNoCallerFunction(const Value *val)
```
Comment (yuleisui) [inline lib/SVF-FE/SymbolTableBuilder.cpp:212]: merge these three functions into a `collectSpecialSym(const Value* val)` and invoked by `collectSym `

**Pull request closed without merging**